roxie/ccd/ccdserver.cpp

```diff
@@ -19164,6 +19164,93 @@ IRoxieServerActivityFactory *createRoxieServerCatchActivityFactory(unsigned _id,
 
 //=================================================================================
 
+class CRoxieServerPullActivity : public CRoxieServerActivity
+{
+    ConstPointerArray buff;
+    bool started;
+    unsigned index;
+
+public:
+    CRoxieServerPullActivity(const IRoxieServerActivityFactory *_factory, IProbeManager *_probeManager)
+        : CRoxieServerActivity(_factory, _probeManager)
+    {
+        started = false;
+        index = 0;
+    }
+
+    virtual void start(unsigned parentExtractSize, const byte *parentExtract, bool paused)
+    {
+        started = false;
+        index = 0;
+        CRoxieServerActivity::start(parentExtractSize, parentExtract, paused);
+    }
+
+    virtual void reset()
+    {
+        while (buff.isItem(index))
+            ReleaseRoxieRow(buff.item(index++));
+        buff.kill();
+        started = false;
```

Comment:
trivial: if you reset started (also set in start()), shouldn't you also reset index = 0 to be consistent?  Or reset neither here.